### PR TITLE
pid1: allow creating scope units based on pidfds (as opposed to numeric pids)

### PR DESCRIPTION
**Pull request by poettering**
One more step towards complete pidfdfication.

### CODE REVIEW

File: src/core/dbus-scope.c [hunk 1]
```diff
@@ -140,8 +140,54 @@ static int bus_scope_set_transient_property(
...
+                }
+
+                r = sd_bus_message_exit_container(message);
+                if (r < 0)
+                        return r;
+
+                if (n <= 0)
+                        return -EINVAL;
+
+                return 1;
```
Comment (daandemeyer) [inline src/core/dbus-scope.c]: I guess this could be merged into a single line with a ternary

**Review by DaanDeMeyer — APPROVED** [review]

**Comment by bluca** [comment]
```
../src/core/dbus-scope.c:146:62: error: unused variable 'creds' [-Werror,-Wunused-variable]
                _cleanup_(sd_bus_creds_unrefp) sd_bus_creds *creds = NULL;
```

**Review by bluca — APPROVED** [review]

File: src/login/logind-dbus.c [hunk 2]
```diff
@@ -3946,7 +3946,7 @@ static int strdup_job(sd_bus_message *reply, char **job) {
 int manager_start_scope(
                 Manager *manager,
                 const char *scope,
-                pid_t pid,
+                PidRef *pidref,
```
Comment (YHNdnzj) [inline src/login/logind-dbus.c]: Hmm, maybe make this `const`?

File: src/shared/bus-unit-util.c [hunk 3]
```diff
@@ -2817,6 +2817,22 @@ int bus_append_unit_property_assignment_many(sd_bus_message *m, UnitType t, char
         return 0;
 }
 
+int bus_append_scope_pidref(sd_bus_message *m, PidRef *pidref) {
```
Comment (YHNdnzj) [inline src/shared/bus-unit-util.c]: (also here)

**Review by YHNdnzj — APPROVED** [review]

**Pull request merged**